### PR TITLE
[CRIMAPP-114-117] Income rehydration and submission

### PR DESCRIPTION
**Pull request by mseedat-moj**
## Description of change
Submits and rehydrates Income Payments and Income Benefits

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-117
https://dsdmoj.atlassian.net/browse/CRIMAPP-114
https://dsdmoj.atlassian.net/browse/CRIMAPP-216
https://dsdmoj.atlassian.net/browse/CRIMAPP-215

## Notes for reviewer
- because of the way `:pence` value types are auto serialised/deserialised into integer pennies, I have had to manually convert the pennies back into pounds before creating the corresponding `IncomeBenefit` and `IncomePayment`. This will also happen with the `OutgoingPayment`: is this the right approach (see new `Payable` module)?
- 

## Screenshots of changes (if applicable)
N/A

## How to manually test the feature
Complete an application with income payments/income benefits and submit the application.

### CODE REVIEW

File: spec/attributes/type/pence_spec.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@
     describe 'when value is an integer' do
       let(:value) { 123 }
 
-      it { is_expected.to eq(12_300) }
+      it { is_expected.to eq(123) }
```
Comment (timpeat) [inline spec/attributes/type/pence_spec.rb:44]: 👍

**Review by timpeat — APPROVED** [review]
Looks great. Thanks for fixing the serialisation issue.

**Review by arthurashman — APPROVED** [review]
All working for me, rehydrates etc. No repeat multiply of pence. \

Do you have another ticket/PR to get the values into the CYA page?

**Comment by mseedat-moj** [comment]
> Looks great. Thanks for fixing the serialisation issue.

You fixed it!!! Take some credit @timpeat !

**Pull request merged**